js/ui/discoveryFeedButton.js

```diff
@@ -54,21 +50,54 @@ function maybeCreateButton() {
  */
 const DiscoveryFeedButton = new Lang.Class({
     Name: 'DiscoveryFeedButton',
-    Extends: St.Button,
+    Extends: St.BoxLayout,
 
     _init: function() {
-        this.parent({ name: 'discovery-feed',
-                      child: new St.Icon({ icon_name: 'go-down-symbolic',
-                                           icon_size: 16,
-                                           style_class: 'discovery-feed-icon ' }),
-                      style_class: 'discovery-feed-button',
+        this.parent({ vertical: true,
                       visible: _primaryMonitorWidthPassesThreshold() });
 
+        this._bar = new St.Button({ name: 'discovery-feed-bar',
+                                    child: new St.Icon({ style_class: 'discovery-feed-bar-icon' }),
+                                    style_class: 'discovery-feed-bar',
+                                    reactive: true  });
+        this.add(this._bar);
+
+        this._tile = new St.Button({ name: 'discovery-feed-tile',
+                                     child: new St.Icon({ style_class: 'discovery-feed-tile-icon' }),
+                                     style_class: 'discovery-feed-tile' });
+        this.add(this._tile, { x_fill: false,
+                               x_align: St.Align.MIDDLE,
+                               expand: true });
+
+        this._bar.connect('event', Lang.bind(this, this._onEvent));
+        this._tile.connect('event', Lang.bind(this, this._onEvent));
+
+        this._bar.connect('notify::hover', Lang.bind(this, this._onHoverChanged));
+        this._tile.connect('notify::hover', Lang.bind(this, this._onHoverChanged));
```

Comment:
For this event and the above, is it possible to connect to the signal on the parent container rather than on both the bar and the tile independently? It seems like clicking on either one should have the same behaviour.

Reply:
Not really. We want the hover on those specific items, the container would include the to the user invisible parts as the hover area. That is why we have the separation between tile and bar actually.